### ISSUE TITLE
Implementation of vimgrep

### ISSUE THREAD

**Issue by TryerGit** [VSCodeVim/Vim#5991]
Hello,

I have a project structure thus:

```
/myproject/
    ----src/
        ----file.cpp
    ----include/
        ----file.h
```

I open up `code .` from `/myproject/`

With `file.cpp` open in the editor window, I am trying to run a vimgrep command thus:

`vimgrep search_item **/*`

On hitting enter, no search is returned, no quick fix list seems to be populated. This happens even though `search_item` features quite prominently in my code. Also, running the above command seems to make the file into a changed state (with no visible change to the file, yet the file "changes" / becomes dirty) requiring me to either save it explicitly or else undo multiple times until it gets back to the original state.

On the bottom status line, I obtain

"E37: No write since last change (add ! to override)"

Instead of VSCode, if I open up `vim` in the same folder, and run the same command, the quick fix list is populated and I am able to traverse this using `:cnext`, `:cprev`, etc.

I searched this forum for previous issues related to `vimgrep` but I could not find any. Hence this OP.

Is there something I am doing wrong? The above is after enabling `NeoVim`.

Thanks.

**Comment by sql-koala** [comment]
> Is there something I am doing wrong?

Not really. But vimgrep is not implemented and doing so would be  a lot of "reinventing the wheel". Because vscode has a reasonably powerful project-wide search feature.
My personal advice is to learn and use vscodes ide features and add vim modal editing on top.

**Comment by J-Fields** [comment]
Re-opening because this is a reasonable request, though I agree VSCode's implementation is probably the way to go
